dask_expr/_collection.py

```diff
@@ -2772,7 +2807,53 @@ def elemwise(op, *args, meta=no_default, out=None, transform_divisions=True, **k
     # TODO: Add align partitions
 
     dfs = [df for df in args if isinstance(df, FrameBase)]
+    if len(dfs) <= 1 or expr.are_co_aligned(*dfs, allow_broadcast=False):
+        result = new_collection(
+            expr.UFuncElemwise(dfs[0], op, meta, transform_divisions, kwargs, *args)
+        )
+    else:
+        result = new_collection(expr.UFuncAlign(dfs[0], op, meta, kwargs, *args))
+    return handle_out(out, result)
 
-    return new_collection(
-        expr.UFuncElemwise(dfs[0], op, meta, transform_divisions, kwargs, *args)
-    )
+
+def handle_out(out, result):
+    """Handle out parameters
+
+    If out is a dask.DataFrame, dask.Series or dask.Scalar then
+    this overwrites the contents of it with the result
+    """
```

Comment:
I feel like this could use a bit more explanation. I'm at least confused. :sweat_smile: